osquery/sql/virtual_table.cpp

```diff
@@ -418,6 +418,25 @@ static int xBestIndex(sqlite3_vtab* tab, sqlite3_index_info* pIdxInfo) {
     cost += 200;
   }
 
+  UsedColumns colsUsed;
+  if (pIdxInfo->colUsed > 0) {
+    for (size_t i = 0; i < columns.size(); i++) {
+      // Check whether the column is used. colUsed has one bit for each of the
+      // first 63 columns, and the 64th bit indicates that at least one other
+      // column is used.
+      uint64_t flag;
+      if (i < 63) {
+        flag = 1L << i;
+      } else {
+        flag = 1L << 63;
```

Comment:
This mimics the example expression here: https://www.sqlite.org/c3ref/index_info.html but the wording mentions "is also set" so is bit 63 and the i - 64'th bit set for column indexes > 64? We do not have tables this wide, so it's not a big deal. We can also add a compile error if someone defines a table with >64 columns, that would be crazy.

My concern is around `*`, does this copy all column names into this structure for normal `select *` queries? If so, can we shortcut this and have a simple indicator "need them all". What do you think, unneeded optimization?

Reply:
Re the "is also set" I think that's just a quirk of phrasing. If it were as you describe that would change the meaning of the other bits (i.e., if bit 1 and 63 were set, you'd need both column 1 and 64?)

Yes, for * this would put all names into the set. Given the # of columns in our tables, it's probably not a big deal.